model/task_lifecycle.go

```diff
@@ -1164,6 +1164,16 @@ func updateBuildStatus(b *build.Build) (bool, error) {
 
 	event.LogBuildStateChangeEvent(b.Id, buildStatus.status)
 
+	notBlockedOrUnscheduled := !buildStatus.allTasksBlocked && !buildStatus.allTasksUnscheduled
```

Comment:
Not necessarily, it can have a task that is blocked but activated, like if a dependency is unattainable.

Reply:
(so having one activated task wouldn't be enough)

Reply:
I see, so if `!buildStatus.allTasksBlocked` means the build has at least one unblocked task and `!buildStatus.allTasksUnscheduled` means the build has at least one scheduled task, then this line is checking if we have at least one **unblocked** activated task, correct?

Reply:
Just saying this because the name of the variable is a little challenging to understand at first glance, and if we can rename it to something more direct.

Reply:
I renamed it to shouldActivate.